### PR TITLE
Add nbQA for notebook and docs linting

### PR DESCRIPTION
**Pull request by basnijholt**
## Description

This is now possible after https://github.com/nbQA-dev/nbQA/pull/745 which solved this issue (https://github.com/nbQA-dev/nbQA/issues/668) I opened a year ago.

## Checklist

- [ ] Fixed style issues using `pre-commit run --all` (first install using `pip install pre-commit`)
- [ ] `pytest` passed

## Type of change

*Check relevant option(s).*

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] (Code) style fix or documentation update
- [ ] This change requires a documentation update

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -25,3 +25,14 @@ repos:
     rev: 3.9.2
     hooks:
       - id: flake8
+  - repo: https://github.com/nbQA-dev/nbQA
+    rev: 1.5.0
+    hooks:
+      - id: nbqa-black
+        additional_dependencies: [jupytext, black]
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:32]: you probably also want
```
  types_or: [jupyter, md]
  types: [text]  # overwrite types: [jupyter]
```

I'll fix this upstream, will push out a 1.5.1

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -25,3 +25,14 @@ repos:
     rev: 3.9.2
     hooks:
       - id: flake8
+  - repo: https://github.com/nbQA-dev/nbQA
+    rev: 1.5.0
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:29]: > I'll fix this upstream, will push out a 1.5.1

done 🚀 

should be enough to just do:
```suggestion
    rev: 1.5.1
```

Comment (basnijholt) [reply .pre-commit-config.yaml]: Thanks! I see that 1.7.0 is already out :)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (basnijholt) [reply .pre-commit-config.yaml:24]: Is this still required?

I notice that all the .md Myst Jupytext notebooks are skipped still.

Comment (basnijholt) [reply .pre-commit-config.yaml:24]: Is this still required?

I notice that all the .md Myst Jupytext notebooks are skipped still.

Comment (MarcoGorelli) [reply .pre-commit-config.yaml:24]: I'm surprised, I've put `types_or: [jupyter, markdown]` in all the hooks

https://github.com/nbQA-dev/nbQA/blob/main/.pre-commit-hooks.yaml

Comment (basnijholt) [reply .pre-commit-config.yaml:24]: Oh actually the pre-commit CI shows that it does work. Not sure why locally these changes weren’t made.

**Pull request merged**